MaxText/inference_microbenchmark.py

```diff
@@ -170,10 +170,20 @@ def collate_results(config, results, model_size, cache_size, num_model_params, i
   return results
 
 
-def write_results(results, filename):
+def write_results(results, filename, flatten_microbenchmark_results):
+  """Write the results microbenchmark results to a json file."""
+  if flatten_microbenchmark_results:
+    flattened_results = {}
+    for key, value in results.items():
+      if isinstance(value, dict):
+        flattened_results.update(value)
+      else:
+        flattened_results[key] = value
+    results = flattened_results
```

Comment:
here we can add the `flattened_results` into existing `results`

Reply:
Yes, I can make the change.

Reply:
Done